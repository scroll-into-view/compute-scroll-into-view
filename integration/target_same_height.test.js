beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/target_same_height')
})

describe('the "nearest" option works with target and frame having equal dimensions', () => {
  test('block: "nearest"', async () => {
    expect.assertions(3)
    const actual = await page.evaluate(() => {
      return window
        .computeScrollIntoView(document.querySelector('.target'), {
          block: 'nearest',
        })
        .map(window.mapActions)
    })
    expect(actual).toHaveLength(2)
    expect(actual[0]).toMatchObject({ el: 'div.container', left: 0, top: 100 })
    expect(actual).toMatchSnapshot()
  })
})
