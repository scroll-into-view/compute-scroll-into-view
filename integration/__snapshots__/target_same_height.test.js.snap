// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the "nearest" option works with target and frame having equal dimensions block: "nearest" 1`] = `
Array [
  Object {
    "el": "div.container",
    "left": 0,
    "top": 100,
  },
  Object {
    "el": "html",
    "left": 0,
    "top": 0,
  },
]
`;
